imageloading-core/src/main/java/dev/chrisbanes/accompanist/imageloading/ImageLoad.kt

```diff
@@ -68,78 +64,43 @@ fun <R : Any, TR : Any> ImageLoad(
     onRequestCompleted: (ImageLoadState) -> Unit = EmptyRequestCompleteLambda,
     content: @Composable BoxScope.(imageLoadState: ImageLoadState) -> Unit
 ) {
-    var state by remember(requestKey) {
-        // We start from the loading state, to avoid thrashing from empty -> loading straightaway
-        mutableStateOf<ImageLoadState>(ImageLoadState.Loading)
-    }
-
-    val callback by rememberUpdatedState(onRequestCompleted)
-
-    val requestActor = remember(requestKey) {
-        ImageLoadRequestActor(executeRequest)
-    }
+    val updatedOnRequestCompleted by rememberUpdatedState(onRequestCompleted)
+    val updatedTransformRequestForSize by rememberUpdatedState(transformRequestForSize)
+    val updatedExecuteRequest by rememberUpdatedState(executeRequest)
 
-    LaunchedEffect(requestActor) {
-        // Launch the Actor
-        requestActor.run { _, newState ->
-            // Update the result state
-            state = newState
+    var requestSize by remember(requestKey) { mutableStateOf<IntSize?>(null) }
 
-            if (newState is ImageLoadState.Success || newState is ImageLoadState.Error) {
-                callback(newState)
-            }
-        }
+    val loadState by produceState<ImageLoadState>(
+        initialValue = ImageLoadState.Loading,
+        key1 = requestKey,
+        key2 = requestSize,
+        key3 = updatedTransformRequestForSize,
```

Comment:
I don't think you need `key3`. Since `updatedTransformRequestForSize` is backed by `rememberUpdatedState`, any update to `transformRequestForSize` won't cause a recomposition here

Reply:
Good point, removed.